modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SExpression.java

```diff
@@ -72,6 +72,7 @@ StatementExpressionNode write(ClassNode classNode) {
 
         statementExpressionNode.setLocation(location);
         statementExpressionNode.setMethodEscape(methodEscape);
+        statementExpressionNode.setNoop(false);
```

Comment:
Why is this necessary?

Reply:
After speaking with you (@stu-elastic) and @rjernst I've decided to change this a bit. I've removed `noop` in favor of `doPop` which will pop extraneous values off the stack when necessary such as a method call with side effects, but an ignored return value `somecall();`.

I also removed the need for StatementExpressionNode to handle return values. Instead, SExpression will just directly translate to a ReturnNode if a return is required. This moves the responsibility to where is makes sense instead of having redundant returns in multiple nodes.

Reply:
I continued to iterate on this after more discussion with @rjernst . I removed the doPop and will instead use an expression's type to determine if a pop is required or not (popping the void type is the same as a noop in this case).